datatypes/ring_programs.py

```diff
@@ -500,69 +505,73 @@ def preprocess(self, mass, circumference, interp_time = None,
                 _interp_time = 0
             else:
                 _interp_time = interp_time
-            interp_time = lambda x: x + _interp_time
-        
+
+            def interp_time(x): return x + _interp_time
+
         if self.timebase == 'by_time':
             if t_start < self[0, 0, 0]:
-                warnings.warn("t_start too early, starting from " 
+                warnings.warn("t_start too early, starting from "
                               + str(self[0, 0, 0]))
                 t_start = self[0, 0, 0]
-    
+
             if t_end > self[0, 0, -1]:
-                warnings.warn("t_stop too late, ending at " 
-                              + str(self[0, 0, -1]))
+                if not np.isinf(t_end):
```

Comment:
No need for double `if`, could just be:
```python
if t_end > self[0, 0, 0] and not np.isinf(t_end):
```

Reply:
You'd still need a second if to catch the case where t_end is np.inf?

Reply:
Ah, misread the indentation.  All good.